File: instrumentation/weblogic-12/src/main/java/com/nr/agent/instrumentation/weblogic12/ResponseWrapper.java
/*
 *
 *  * Copyright 2020 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.agent.instrumentation.weblogic12;

import weblogic.servlet.internal.ServletResponseImpl;

import com.newrelic.api.agent.HeaderType;
import com.newrelic.api.agent.Response;

public class ResponseWrapper implements Response {

    private final ServletResponseImpl delegate;

    public ResponseWrapper(ServletResponseImpl servletResponse) {
        this.delegate = servletResponse;
    }

    @Override
    public int getStatus() throws Exception {
        return delegate.getStatus();
    }

    @Override
    public String getStatusMessage() throws Exception {
        return null;
    }

    @Override
    public void setHeader(String name, String value) {
        delegate.setHeader(name, value);
    }

    @Override
    public String getContentType() {
        return delegate.getContentType();
    }

    @Override
    public HeaderType getHeaderType() {
        return HeaderType.HTTP;
    }
}

File: instrumentation/apache-pekko-http-core-2.13_1/src/main/scala/com/nr/instrumentation/pekkohttpcore/ResponseWrapper.scala
/*
 *
 *  * Copyright 2024 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.instrumentation.pekkohttpcore

import org.apache.pekko.http.scaladsl.model.HttpResponse
import org.apache.pekko.http.scaladsl.model.headers.RawHeader
import com.newrelic.api.agent.{HeaderType, ExtendedResponse}

class ResponseWrapper(var response: HttpResponse) extends ExtendedResponse {

  def getStatus: Int = {
    response.status.intValue
  }

  def getStatusMessage: String = {
    response.status.reason
  }

  def getContentType: String = {
    response.entity.contentType.value
  }

  override def getHeaderType = HeaderType.HTTP

  override def setHeader(name: String, value: String): Unit = {
    response = response.addHeader(new RawHeader(name, value))
  }

  def getContentLength: Long = {
    response.entity.contentLengthOption.getOrElse(0L)
  }
}
